app/src/main/java/com/skyyo/samples/features/videopager/VideoPagerScreen.kt

```diff
@@ -0,0 +1,141 @@
+package com.skyyo.samples.features.videopager
+
+import android.net.Uri
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.gestures.Orientation
+import androidx.compose.foundation.interaction.MutableInteractionSource
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.Icon
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.PlayArrow
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.viewinterop.AndroidView
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.google.accompanist.pager.ExperimentalPagerApi
+import com.google.android.exoplayer2.C
+import com.google.android.exoplayer2.ExoPlayer
+import com.google.android.exoplayer2.MediaItem
+import com.google.android.exoplayer2.Player
+import com.google.android.exoplayer2.source.ProgressiveMediaSource
+import com.google.android.exoplayer2.ui.AspectRatioFrameLayout
+import com.google.android.exoplayer2.ui.PlayerView
+import com.google.android.exoplayer2.upstream.DataSource
+import com.google.android.exoplayer2.upstream.DefaultDataSource
+import com.google.android.exoplayer2.upstream.DefaultHttpDataSource
+import com.skyyo.samples.features.videopager.composables.VideoPager
+import com.skyyo.samples.features.videopager.model.PagerState
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalPagerApi::class)
+@Composable
+fun VideoPagerScreen(viewModel: VideoPagerViewModel = hiltViewModel()) {
+
+    val videoUrls by viewModel.videos.collectAsState()
+
+    ShortViewCompose(videoItemsUrl = videoUrls, clickItemPosition = 0)
+}
+
+@OptIn(ExperimentalPagerApi::class)
+@Composable
+fun ShortViewCompose(
+    videoItemsUrl: List<String>,
+    clickItemPosition: Int = 0,
+) {
+    val pagerState: PagerState = remember {
+        PagerState(clickItemPosition, 0, videoItemsUrl.size - 1)
+    }
+    val isPauseIconVisible = remember { mutableStateOf(false) }
+    VideoPager(
+        state = pagerState,
+        orientation = Orientation.Vertical,
+        offscreenLimit = 1
+    ) {
+        isPauseIconVisible.value = false
+        VideoPlayer(
+            videoUrl = videoItemsUrl[page],
+            pagerState = pagerState,
+            page = page,
+            isPauseIconVisibleState = isPauseIconVisible
+        )
```

Comment:
this implementation creates multiple instances of ExoPlayer, which shouldn't happen